### PR TITLE
[TASK] Add template type option for init command

### PR DESCRIPTION
**Pull request by sabbelasichon**
Resolves: #6232

### CODE REVIEW

File: src/Configuration/Option.php [hunk 1]
```diff
@@ -157,4 +157,9 @@ final class Option
      * @var string
      */
     public const OPTION_NO_DIFFS = 'no-diffs';
+
+    /**
+     * @var string
+     */
+    public const TEMPLATE_TYPE = 'type';
```
Comment (TomasVotruba) [inline src/Configuration/Option.php:164]: Value and name should be equal. That way we can avoid conflict with e.g. `ANY_TYPE = 'type';`

**Comment by TomasVotruba** [comment]
The docs file should be added as well

File: src/Configuration/Option.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Configuration/Option.php:164]: So it should be template-type with or without prefix OPTION?

Comment (TomasVotruba) [reply src/Configuration/Option.php:164]: What would you prefer?

Comment (sabbelasichon) [reply src/Configuration/Option.php:164]: For the value template-type and for the constant name i would use TEMPLATE_TYPE also. Not using OPTION prefix here. But sometimes it is used. Do not see a pattern where and why.

**Comment by sabbelasichon** [comment]
> The docs file should be added as well
What do you mean with docs file? The only place where the init command is mentioned right now is in the README.md. 
As there is no other official template-type for now i would prefer to drop it at the beginning and add it to the docs once there are different template files.

File: src/Configuration/Option.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Configuration/Option.php:164]: I see. The option prefix is for the command options. So we should add it here as well. Sorry too early and yesterday too late

Comment (TomasVotruba) [reply src/Configuration/Option.php:164]: No troubles :)

Personally I would go for:

```php
const TEMPLATE_TYPE = 'template_type'
```

**Comment by TomasVotruba** [comment]
> What do you mean with docs file?

A small documentation, that describes this feature. E.g. in `/docs/init_command.md`
The goal is to other people understand without looking at this PR and its description.

File: src/Configuration/Option.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Configuration/Option.php:164]: Ok. But the pattern for options within a command is different. For existing ones it is always OPTION_FOO_BAR = 'option-with-dash'

Comment (TomasVotruba) [reply src/Configuration/Option.php:164]: CLI is only one of possible inputs. The parameter can come from manual parameter too.

**Comment by sabbelasichon** [comment]
I think we should create an issue for the FileProcessors documentation. But before documenting it i would prefer to streamline the API a little bit like skipping them or running them without a Rector should be possible.

File: src/Configuration/Option.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Configuration/Option.php:164]: But in cli mode i would prefer the dash. And this is visible for the user. In all other parts it is a hidden detail. Naming things is still hard and not solved by robots till now ;-)

File: docs/init_command.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    $services->set(TypedPropertyRector::class);
+};
+```
+
+The init command takes an option called --template-type or in short form -t.
+If some other Rector extension like [rector-nette](https://github.com/rectorphp/rector-nette) or [rector-doctrine](https://github.com/rectorphp/rector-doctrine) provides a custom template type you can specify it here:
+
+```bash
+vendor/bin/rector init --template-type=nette
+```
```
Comment (TomasVotruba) [inline docs/init_command.md:34]: I wonder, how can I add such a template as a developer? 

When should I use this extra option and when only `init`?

Comment (sabbelasichon) [reply docs/init_command.md]: I enhanced the documentation to explain this part also. Thanks for asking.

File: docs/init_command.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+If you just want to use the default template provided by Rector you can omit the --template-type option.
+
+# How to add a template type as a developer
+In order to provide a new template type as a developer you should create a custom template class implementing the TemplateResolverInterface:
+
+```
```
Comment (TomasVotruba) [inline docs/init_command.md:41]:
```suggestion
```php
```

File: docs/init_command.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+The init command takes an option called --template-type or in short form -t.
+If some other Rector extension like [rector-nette](https://github.com/rectorphp/rector-nette) or [rector-doctrine](https://github.com/rectorphp/rector-doctrine) provides such a custom template type you can specify it here:
+
+```bash
+vendor/bin/rector init --template-type=nette
+```
+
+If you just want to use the default template provided by Rector you can omit the --template-type option.
+
+# How to add a template type as a developer
```
Comment (TomasVotruba) [inline docs/init_command.md:38]: :+1:

File: docs/init_command.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    $parameters = $containerConfigurator->parameters();
+    $parameters->set(Option::SETS, [SetList::CODE_QUALITY]);
+
+    // register single rule
+    $services = $containerConfigurator->services();
+    $services->set(TypedPropertyRector::class);
+};
+```
+
+The init command takes an option called --template-type or in short form -t.
```
Comment (TomasVotruba) [inline docs/init_command.md:29]:
```suggestion
The init command takes an option called `--template-type`.
```

Comment (TomasVotruba) [reply docs/init_command.md:29]: Adding short options makes command less readable, so I would prefer avoid it untill it's something generic across CLI apps e.g. like `-n` for `--dry-run`.

File: docs/init_command.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+If some other Rector extension like [rector-nette](https://github.com/rectorphp/rector-nette) or [rector-doctrine](https://github.com/rectorphp/rector-doctrine) provides such a custom template type you can specify it here:
+
+```bash
+vendor/bin/rector init --template-type=nette
+```
```
Comment (TomasVotruba) [inline docs/init_command.md:34]: Thanks for completing the answers!

One last detail - what is the motivation to use a "nette" template for `rector.php`? How does it differ from the main `rector.php`? So people understand, why they should consider using the `nette` one over generic one.

File: src/Console/Command/InitCommand.php [hunk 7]
```diff
@@ -29,26 +33,45 @@ final class InitCommand extends Command
...
+        $this->templateResolvers = $templateResolvers;
         parent::__construct();
     }
 
     protected function configure(): void
     {
         $this->setDescription('Generate rector.php configuration file');
+        $this->addOption(
+            Option::TEMPLATE_TYPE,
+            't',
```
Comment (TomasVotruba) [inline src/Console/Command/InitCommand.php:62]:
```suggestion
            null,
```

File: docs/init_command.md (hunk 6 above)
Comment (sabbelasichon) [reply docs/init_command.md:34]: Could we answer this later? Because i don´t know yet. I only know my personal motivation for TYPO3.

Comment (sabbelasichon) [reply docs/init_command.md:34]: I would have to come up with a contrived example because no template for nette or any other extension exists yet.

Comment (sabbelasichon) [reply docs/init_command.md:34]: I promise i will do that later.

Comment (TomasVotruba) [reply docs/init_command.md:34]: > I only know my personal motivation for TYPO3

Real motivation is the best. Put it there :+1:

Comment (sabbelasichon) [reply docs/init_command.md]: Done

Comment (TomasVotruba) [reply docs/init_command.md]: Awesome, now I get it! Thanks :+1:

**Pull request merged**